java/client/src/test/java/glide/api/RedisClientTest.java

```diff
@@ -53,21 +68,239 @@ public void customCommand_interruptedException() throws ExecutionException, Inte
         String key = "testKey";
         Object value = "testValue";
         String cmd = "GETSTRING";
+        String[] arguments = new String[] {cmd, key};
         CompletableFuture<Object> testResponse = mock(CompletableFuture.class);
         InterruptedException interruptedException = new InterruptedException();
         when(testResponse.get()).thenThrow(interruptedException);
-        when(commandManager.submitNewCommand(any(), any())).thenReturn(testResponse);
+
+        // match on protobuf request
+        when(commandManager.submitNewCommand(any(), any(), any(), any())).thenReturn(testResponse);
 
         // exercise
         InterruptedException exception =
                 assertThrows(
                         InterruptedException.class,
                         () -> {
-                            CompletableFuture<Object> response = service.customCommand(new String[] {cmd, key});
+                            CompletableFuture<Object> response = service.customCommand(arguments);
                             response.get();
                         });
 
         // verify
         assertEquals(interruptedException, exception);
     }
+
+    @Test
+    public void ping_success() throws ExecutionException, InterruptedException {
+        // setup
+        CompletableFuture<String> testResponse = mock(CompletableFuture.class);
+        when(testResponse.get()).thenReturn("PONG");
+
+        // match on protobuf request
+        when(commandManager.<String>submitNewCommand(any(), any(), any(), any()))
+                .thenReturn(testResponse);
+
+        // exercise
+        CompletableFuture<String> response = service.ping();
+        String payload = response.get();
+
+        // verify
+        assertEquals(testResponse, response);
+        assertEquals("PONG", payload);
+    }
+
+    @Test
+    public void pingWithMessage_success() throws ExecutionException, InterruptedException {
```

Comment:
Please update all new test names
```suggestion
    @SneakyThrows
    public void ping_with_message_success() {
```